doc/source/serve/tutorials/gradio-integration.md

```diff
@@ -0,0 +1,137 @@
+
+We will be using HuggingFace's pipeline to access the model. First, let's install the transformers module.
+```bash
+$ pip install transformers
+```
+
+## Basic Example: Deploy your Gradio App with Ray Serve
+
+In this example, we will show you the easy, straightforward way to deploying your app onto Ray. Start by creating a new python file named `demo.py` and import all dependencies.
+```{literalinclude} ../../../../python/ray/serve/examples/doc/gradio-integration.py
+:start-after: __doc_import_begin__
+:end-before: __doc_import_end__
+```
+
+Then, we construct the (optional) Gradio app `io`:
+> **Note**: Remember you can substitute this with your own Gradio app if you want to try scaling up your own Gradio app!
+```{literalinclude} ../../../../python/ray/serve/examples/doc/gradio-integration.py
+:start-after: __doc_gradio_app_begin__
+:end-before: __doc_gradio_app_end__
+```
+
+
+### Understanding `GradioServer`
+In order to deploy your Gradio App onto a Ray Cluster, you need to wrap your Gradio App in a serve deployment. `GradioIngress` acts as that wrapper. It serves your Gradio App on a Ray cluster.
+```python
+@serve.deployment
+class GradioServer:
+    def __init__(self, io: gr.Blocks):
+        self.app = gr.routes.App.create_app(io)
+
+    async def __call__(self, request: starlette.requests.Request):
+        sender = ASGIHTTPSender()
+        await self.app(request.scope, receive=request.receive, send=sender)
+        return sender.build_asgi_response()
+```
+
+### Deploy your Gradio Server
+You can increase the number of replicas of your application (which can take on more client requests) or increase the number of CPUs and/or GPUs available to each replica.
+
+Then, using either the example we created above, or an existing Gradio App (of type `Interface`, `Block`, `Parallel`, etc.), wrap it in your Gradio Server.
+
+```{literalinclude} ../../../../python/ray/serve/examples/doc/gradio-integration.py
+:start-after: __doc_app_begin__
+:end-before: __doc_app_end__
+```
+
+Finally, deploy your Gradio Server!
+```bash
+$ serve run demo:app
+```
+
+Now you can access your Gradio app at `http://localhost:8000`! This is what it should look like:
+![Gradio Result](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/gradio_result.png)
+
+
+## Complex Example: Parallelism
+If you have multiple models you want to run in parallel, one way to do so is directly through Ray Serve.
+
+### Original Approach
+Suppose you want to run the following program.
+
+1. Take two text generation models, `gpt2` and `EleutherAI/gpt-neo-125M`.
+2. Run the two models on the same input text, such that the generated text has a minimum length of 20 and maximum length of 100.
+3. Display the outputs of both models using Gradio.
+
+This is how you would do it normally:
+
+```python
```

Comment:
I played around with it and there doesn't seem to be a good way to test the `launch()` statement, as it will block the python script and the test statements afterwards don't get run. Synced with @shrekris-anyscale and decided not to include for now.